devnet-sdk/devstack/stack/testing.go

```diff
@@ -0,0 +1,138 @@
+package stack
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// T is a minimal subset of testing.T
+// This can be implemented by tooling, or a *testing.T can be used directly.
+// The T interface is only used in the stack package for sanity-checks,
+// where local failure there and then is preferable over bubbling up the error.
+type T interface {
+	Errorf(format string, args ...interface{})
+	FailNow()
+	TempDir() string
+	Cleanup(fn func())
+	Logf(format string, args ...any)
+	Helper()
+	Name() string
+}
+
+// This testing subset is sufficient for the require.Assertions to work.
+var _ require.TestingT = T(nil)
+
+// ToolingT is a T implementation that can be used in tooling,
+// when the devnet-SDK is not used in a regular Go test.
+type ToolingT struct {
+	// TestName, for t.Name() purposes
+	TestName string
+
+	// Errors will be logged here.
+	Log log.Logger
+
+	// Fail will be called to register a critical failure.
+	// The implementer can choose to panic, crit-log, exit, etc. as preferred.
+	Fail func()
+
+	// cleanup stack
+	cleanupLock    sync.Mutex
+	cleanupBacklog []func()
+}
+
+var _ T = (*ToolingT)(nil)
+
+func (t *ToolingT) Errorf(format string, args ...interface{}) {
+	t.Log.Error(fmt.Sprintf(format, args...))
+}
+
+func (t *ToolingT) FailNow() {
+	t.Fail()
+}
+
+func (t *ToolingT) TempDir() string {
+	// The last "*" will be replaced with the random temp dir name
+	tempDir, err := os.MkdirTemp("", "op-dev-*")
+	if err != nil {
+		t.Errorf("failed to create temp dir: %v", err)
+		t.FailNow()
+	}
+	require.NotEmpty(t, tempDir, "sanity check temp-dir is not empty")
```

Comment:
```suggestion
	require.NotEmpty(t, tempDir, "sanity check temp-dir path is not empty")
```

The directory should be empty, the path shouldn't be - had me confused for a bit.